### PR TITLE
Epub accessibility check

### PR DESCRIPTION
**Pull request by mwbenowitz**
This adds a new step (and tests) to the s3 storage process that generates an accessibility report for all ePubs being stored. This report is generated through the Daisy Consortium's Ace Report tool, which is running behind an API on an ec2 instance.

There are various reasons why this is running on a standalone instance, the main being that Ace includes Chromium as a dependency and it is difficult/inadvisable to run a Lambda which includes chromium. 

This branch adds a new output object to the Kinesis stream that includes the full Report data and summaries, this is to be read by the database updater add to add to a new table in the ResearchNow data model.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,71 +1,109 @@
 import axios from 'axios'
-import LambdaEnvVars from 'lambda-env-vars'
-import {checkForExisting, epubStore, epubExplode, getBuffer} from './src/epubParsers'
-import {resultHandler} from './src/responseHandlers'
+import AccessibilityChecker from './src/accessibilityCheck'
+import Parser from './src/epubParsers'
+import ResHandler from './src/responseHandlers'
 
 const fileNameRegex = /[0-9]+[.]{1}epub[.]{1}(?:no|)images/
```
Comment (gkallenberg) [inline index.js:8]: wondering if the `[.]` should be escaped as in `[\.]` if the period or dot is what is being tested.

File: src/epubParsers.js [hunk 2]
```diff
@@ -1,113 +1,110 @@
...
 import unzip from 'unzip-stream'
-import stream from 'stream'
-import {resultHandler} from './responseHandlers'
+import ResHandler from './responseHandlers'
 
 AWS.config.update({
-    region: 'us-east-1',
-    logger: process.stdout
+  region: 'us-east-1',
+  logger: process.stdout
```
Comment (gkallenberg) [inline src/epubParsers.js]: we'll need to add a logger like winston for sending standard log messages to CloudWatch

File: test/accessibilityCheck.test.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      let resp
+      try{
+        resp = await AccessibilityChecker.getAccessibilityReport({buf: 'fakeBuf'})
+      } catch(err) {
+        resp = err
+      }
+      expect(resp.data).to.equal('Some Test Data')
+    })
+
+    it('should get a throw an error if it does not get a report', async () => {
```
Comment (gkallenberg) [inline test/accessibilityCheck.test.js]: edit: `should throw an error...`

File: test/handler.test.js [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+      handlerStub = sinon.stub(ResHandler, 'resultHandler')
+      parseStub = sinon.stub(Lambda, 'parseRecords')
+    })
+
+    afterEach(() => {
+      handlerStub.restore()
+      parseStub.restore()
+    })
+
+    it('should return 500 if no records in array', () => {
```
Comment (gkallenberg) [inline test/handler.test.js]: Is an empty array indicating no records were found? It's a valid array object, but without a set of records to process to processing should end. I think we talked about this in stand up -- using a different status code like a 204 No Content response? That would indicate that the request was fulfilled and terminate the process.

**Review by gkallenberg — COMMENTED** [review]
Looks good. I would add some of the basics from our [documentation standards](https://github.com/NYPL/engineering-general/blob/master/standards/documentation.md) to the README like requirements, dependencies, installation instructions, testing instructions. I need to do this on my repos as well. Let me know if the comments make sense or not.

Check your linter errors.

And we should have a standard logging package like `winston` to handle errors in a standard way for CloudWatch.

Should also talk about retry scenarios for 5xx responses since servers/services may be unavailable temporarily.

File: index.js [hunk 5]
```diff
@@ -1,71 +1,109 @@
...
+  try{
+    let accessReport = await AccessibilityChecker.getAccessibilityReport(zipData)
+    accessReport['id'] = itemID
+    return {
+      "status": 200,
+      "code": "accessibility",
+      "message": "Created Accessibility Score",
+      "data": accessReport
+    }
+  } catch(err) {
```
Comment (seanredmond) [inline index.js]: I wonder if it's cleaner just to rethrow the error here? And similarly in the filename match below and other places, there are a bunch of spots where a 500 error might be returned, would it be simpler to throw errors and have them caught, logged, and converted to a return code in one place?

File: index.js [hunk 6]
```diff
@@ -1,71 +1,109 @@
...
-            })
-            .catch((error) => {
-                handleResp = {
-                    "status": error.response.status,
-                    "code": "Axios Failure",
-                    "message":error.response.data
-                }
-                resultHandler(handleResp)
-            })
+exports.parseRecord = (record) => {
```
Comment (seanredmond) [inline index.js:62]: This feels like two functions to me. One to parse the JSON, check the filename, and either throw an error or return the filename, id, and date; and another to take those and do the existence checking and exploding.

Comment (mwbenowitz) [reply index.js:62]: I agree, I've split out the functionality into two separate methods. It revealed a few issues with the code and made the test suite coverage much more accurate

File: index.js (hunk 5 above)
Comment (mwbenowitz) [reply index.js]: After doing the refactoring suggested below, this is essentially what it does now. It throws Lambda Errors, which are handled/logged at a higher level

File: test/accessibilityCheck.test.js (hunk 3 above)
Comment (mwbenowitz) [reply test/accessibilityCheck.test.js]: thanks!

File: src/epubParsers.js (hunk 2 above)
Comment (mwbenowitz) [reply src/epubParsers.js]: I've added Winston as the logger for the function but not on this line. I think that this output should be reflected in CloudWatch, but I need to look at how best to integrate winston with the AWS SDK. There is at least one library that does so, but I haven't reviewed it yet

File: index.js (hunk 1 above)
Comment (mwbenowitz) [reply index.js:8]: within the brackets escaping is unnecessary. I can add it in if you think its clearer

**Comment by mwbenowitz** [comment]
Sorry for the delay on responding to these comments! Thanks for taking a look at everything. In addition to the specific comments here I also updated the Readme to have more details from the engineering guidelines and addressed the linter errors as Greg pointed out. I'm merging these changes now because I'm updating the input/response formats to better match the data model that we've now established with the persistence layer and it will make it easier to integrate the response from this function (especially for the reports)

**Pull request merged**